lib/store/store.ts

```diff
@@ -8,13 +8,19 @@ import {
   Task,
   TimeEntry,
 } from "../adapters/types";
-import { isSameDay, startOfToday, subWeeks } from "date-fns";
-import { DateRange } from "@mui/lab";
+import { endOfMonth, isSameDay, startOfMonth, subMonths } from "date-fns";
 import { Alert, AsyncState, StoreOptions } from "./types";
+import { AxiosError } from "axios";
 
 export class Store<S extends AdapterCredentials, T extends AdapterCredentials> {
   alert: Alert | null = null;
-  dateRange: DateRange<Date> = [subWeeks(startOfToday(), 2), startOfToday()];
+  dateRange: [Date, Date] =
+    new Date().getDate() <= 10
+      ? [
+          startOfMonth(subMonths(new Date(), 1)),
+          endOfMonth(subMonths(new Date(), 1)),
+        ]
+      : [startOfMonth(new Date()), endOfMonth(new Date())];
```

Comment:
Please extract it into a function, and make sure to call `new Date()` only once in that function